src/plugins/backend_odbc/backend_odbc_set.c

```diff
@@ -0,0 +1,1231 @@
+					elektraFree (query);
+					elektraFree (queryMetaInsert);
+					return -1;
+				}
+				else if (curAffectedRows > 0)
+				{
+					sumAffectedRows += curAffectedRows;
+				}
+			}
+
+			elektraFree (queryMetaInsert);
+		}
+	}
+
+	elektraFree (query);
+	return sumAffectedRows;
+}
+
+#if DEBUG
+/**
+ * @internal
+ *
+ * @brief Print all keys that was detected as added, removed or modified.
+ * 	Mainly useful for logging and debugging.
+ *
+ * @param diffSet The ElektraDiff that was created by comparing the data to store with the data that is currently in the data source.
+ * @param parentKey The key-name of this key is used to print the relative key-names of the affected Keys.
+ * 	This relative key-names are actually stored in the datasource and enables the support for mounting the data source
+ * 	at different places in the KDB.
+ */
+static void logChangedKeys (ElektraDiff * diffSet, Key * parentKey)
+{
+	ELEKTRA_LOG ("Added keys:\n");
+	KeySet * ksDiff = elektraDiffGetAddedKeys (diffSet);
+	for (elektraCursor it = 0; it < ksGetSize (ksDiff); it++)
+	{
+		Key * curKey = ksAtCursor (ksDiff, it);
+		ELEKTRA_LOG ("keyname: %s, string: %s, relative: %s\n", keyName (curKey), keyString (curKey),
+			     elektraKeyGetRelativeName (curKey, parentKey));
+	}
+	ksDel (ksDiff);
+
+	ELEKTRA_LOG ("Removed keys:\n");
+	ksDiff = elektraDiffGetRemovedKeys (diffSet);
+	for (elektraCursor it = 0; it < ksGetSize (ksDiff); it++)
+	{
+		Key * curKey = ksAtCursor (ksDiff, it);
+		ELEKTRA_LOG ("keyname: %s, string: %s, relative: %s\n", keyName (curKey), keyString (curKey),
+			     elektraKeyGetRelativeName (curKey, parentKey));
+	}
+	ksDel (ksDiff);
+
+	ELEKTRA_LOG ("Modified keys:\n");
+	ksDiff = elektraDiffGetModifiedKeys (diffSet);
+	for (elektraCursor it = 0; it < ksGetSize (ksDiff); it++)
+	{
+		Key * curKey = ksAtCursor (ksDiff, it);
+		ELEKTRA_LOG ("keyname: %s, string: %s, relative: %s\n", keyName (curKey), keyString (curKey),
+			     elektraKeyGetRelativeName (curKey, parentKey));
+	}
+	ksDel (ksDiff);
+}
+#endif
+
+/**
+ * @internal
+ *
+ * @brief An ElektraDiff contains Keys for which the value and/or metadata got changed.
+ * But these Keys have the old values set, we need the new values which should be stored in the data source.
+ *
+ * @param ks The KeySet with the Keys that should be persisted. This Keys must contain the new values.
+ * 	This KeySet must contain Keys with all the key-names that are also stored in @p ksModifiedOldKeys.
+ * 	The KeySet @p ks may contain more keys then @p ksModifiedOldKeys, but not less!
+ * @param ksModifiedOldKeys The KeySet that contains the keys thats got modified.
+ * 	For this function to be useful, this KeySet usually contains the modified Keys with the old values and metadata.
+ *
+ * @return A filtered KeySet with the Keys from @p ks that also were part of @p ksksModifiedOldKeys.
+ * 	The filtering is based on the key-names.
+ */
+static KeySet * getModifiedNewKeys (KeySet * ks, KeySet * ksModifiedOldKeys)
```

Comment:
We actually have an `elektraDiffGetModifiedNewKeys` function, it would be better if you it instead of reimplementing your own version her. (AFAIK, it's still considered private, but that shouldn't be a problem. Either you include the private header here, or you make the function public)

Reply:
Thank you, I had in mind that such a function exists, but wasn't sure why it was private.

I've moved the function to the public header, because this function looks very important to me. Often, you need the new values of modified keys when you calculate an `ElektraDiff`.
Moreover, they are calculated by `elektraDiffCalculate ()` anyway.